test/Dialect/FIRRTL/lower-types.mlir

```diff
@@ -1045,18 +1045,23 @@ firrtl.module private @is1436_FOO() {
 
 firrtl.circuit "NLALowering" {
   // Check if the NLA is updated with the new lowered symbol on a field element.
-  firrtl.hierpath @nla [@fallBackName::@test, @Aardvark::@test, @NLALowering::@b]
-  // CHECK: firrtl.hierpath @nla_0 [@fallBackName::@test, @Aardvark::@test, @NLALowering::@b_data]
-  // CHECK: firrtl.hierpath @nla [@fallBackName::@test, @Aardvark::@test, @NLALowering::@b_ready]
-  firrtl.hierpath @nla_1 [@fallBackName::@test, @Aardvark::@test_1, @NLALowering]
-  firrtl.hierpath @nla_2 [@fallBackName::@test, @Aardvark::@test, @NLALowering::@b2]
+  firrtl.hierpath @nla_b [@fallBackName::@test, @Aardvark::@test, @NLALowering::@b]
+  // CHECK:      firrtl.hierpath @nla_b [@fallBackName::@test, @Aardvark::@test, @NLALowering::@b_valid]
+  // CHECK-NEXT: firrtl.hierpath @nla_b_0 [@fallBackName::@test, @Aardvark::@test, @NLALowering::@b_ready]
+  // CHECK-NEXT: firrtl.hierpath @nla_b_1 [@fallBackName::@test, @Aardvark::@test, @NLALowering::@b_data]
+  firrtl.hierpath @nla [@fallBackName::@test, @Aardvark::@test_1, @NLALowering]
+  // CHECK-NEXT: firrtl.hierpath @nla [@fallBackName::@test, @Aardvark::@test_1, @NLALowering]
+  firrtl.hierpath @nla_b2 [@fallBackName::@test, @Aardvark::@test, @NLALowering::@b2]
+  // CHECK-NEXT: firrtl.hierpath @nla_b2 [@fallBackName::@test, @Aardvark::@test, @NLALowering::@b2_valid]
+  // CHECK-NEXT: firrtl.hierpath @nla_b2_0 [@fallBackName::@test, @Aardvark::@test, @NLALowering::@b2_ready]
+  // CHECK-NEXT: firrtl.hierpath @nla_b2_1 [@fallBackName::@test, @Aardvark::@test, @NLALowering::@b2_data]
```

Comment:
This shows the main difference created by this PR. You get one hierarchical path _for every_ ground type as opposed to one hierarchical path for each NLA. This is a more restrictive, but a more strictly correct implementation of lowering.

Reply:
Is that correct, shoud we get a `HierPathOp` for a ground type that does not have a nonlocal annotation ? The `HierPathOp` seems to be invalid in that case !